logstash-core/src/test/java/org/logstash/log/ListAppender.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.logstash.log;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginBuilderAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginBuilderFactory;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.validation.constraints.Required;
+import org.apache.logging.log4j.core.impl.MutableLogEvent;
+import org.apache.logging.log4j.core.layout.SerializedLayout;
+import org.jruby.embed.internal.BiVariableMap;
+
+/**
+ * This appender is primarily used for testing. Use in a real environment is discouraged as the
+ * List could eventually grow to cause an OutOfMemoryError.
+ *
+ * This appender will use {@link Layout#toByteArray(LogEvent)}.
+ *
+ * see org.apache.logging.log4j.junit.LoggerContextRule#getListAppender(String) ILC.getListAppender
+ */
+@Plugin(name = "List", category = "Core", elementType = "appender", printObject = true)
+public class ListAppender extends AbstractAppender {
```

Comment:
This class is a copy/paste from log4j-core test jar.
It would be nicer to remove this class `ListAppender`, and add this dependency in `build.gradle` :
`testCompile(group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.6.1', classifier: 'tests')`

Thus, it would let to make interesting unit tests on log4j2 integration, using this kind of code :
https://logging.apache.org/log4j/2.x/manual/configuration.html#UnitTestingInMaven